pkg/discovery/summary.go

```diff
@@ -0,0 +1,238 @@
+/*
+Copyright the Sonobuoy contributors 2022
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package discovery
+
+import (
+	"os"
+	"path"
+	"path/filepath"
+	"io/fs"
+        "io/ioutil"
+	v1 "k8s.io/api/core/v1"
+	k8sver "k8s.io/apimachinery/pkg/version"
+	"github.com/sirupsen/logrus"
+	"encoding/json"
+)
+
+type ClusterSummary struct {
+	NodeHealth HealthInfo `json:"node_health" yaml:"node_health"`
+	PodHealth HealthInfo `json:"pod_health" yaml:"pod_health"`
+	APIVersion string `json:"api_version" yaml:"api_version"`
+	ErrorInfo LogSummary `json:"error_summary" yaml:"error_summary"`
+}
+
+
+type HealthInfo struct {
+	Total int `json:"total_nodes" yaml:"total_nodes"`
+	Healthy int `json:"healthy_nodes" yaml:"healthy_nodes"`
+	Details []HealthInfoDetails `json:"details,omitempty" yaml:"details,omitempty"`
+}
+
+type HealthInfoDetails struct {
+	Name string `json:"name" yaml:"name"`
+	Healthy bool `json:"healthy" yaml:"healthy"`
+	Ready string `json:"ready" yaml:"ready"`
+	Reason string `json:"reason,omitempty" yaml:"reason,omitempty"`
+	Message string `json:"message,omitempty" yaml:"message,omitempty"`
+	Namespace string `json:"namespace,omitempty" yaml:"namespace,omitempty"`
+}
+
+const (
+	//Filename of the core nodes json output, relative to ClusterResourceLocation
+	CoreNodesFile = "core_v1_nodes.json"
+
+	//Filename of the core pod json output, relative to NSResourceLocation
+	CorePodsFile = "core_v1_pods.json"
+
+	//Filename for the server version
+	ServerVersionFile = "serverversion.json"
+
+	//Filename of the cluster health, relative to MetaLocation
+	ClusterHealthFile = "clusterhealth.json"
+)
+
+// LoadJsonIntoStruct reads a json file into an object.
+// fileName needs to be the name of a file that can be opened
+// error will be returned if opening the file fails,
+// or if decoding its content as Json into the supplied "object" fails
+func LoadJsonIntoStruct(fileName string, object interface{}) (err error) {
+	file, err := os.Open(fileName)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+	decoder := json.NewDecoder(file)
+	err = decoder.Decode(object)
+	//no need to check for err as we are returning it anyway
+	return err
+}
+
+//Read the server version from the working directory and return it as a string
+func ReadVersion(tarballRootDir string) (string, error) {
```

Comment:
[TODO, we can just make it in a clean-up PR]
It seems really odd that we have to re-impelemnt this (not your fault). We have these sort of methods already defined in `reader.go` but its always been targeted at a tarball and we read it without having to unzip it.

Seems like it would be really simple to update the methods to take a directory or tarball so we can centralize the file structure info/methods.